molgenis/molgenis-emx2
# fix(ontology): disable node select when searching so users don't select hidden terms

in part https://github.com/molgenis/GCC/issues/819
(other half in #4539)

to test:
- when using ontology select in forms and filters it will remove the 'node' selector when searching
- see issue that was mentioned

Spec based on discussion
- [x] if search matches only a leaf and not its parent nodes, we show only the leaf plus its path. Node can not be selected
- [x] if search match only a node and not its child nodes, we show the node and then all its children without search filter

todo:
- [ ] check edge case when the node is found but none of its children, how should it then behave, is select all then ok?
- [ ] test if this leads to incomplete data entry
- [ ] check if there are other applications affected, e.g. directory and if that is desired

N.B. a next step would be to also enable a special ontology type (categorical) that would have node selection always disabled. See #ontology
